### PR TITLE
Add issue labeler tool

### PR DESCRIPTION
**Pull request by trodge**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: tools/issue-labeler/main.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	if err != nil {
+		glog.Exitf("Error unmarshalling enrolled teams yaml: %v", err)
+	}
+	issueBody := os.Getenv("ISSUE_BODY")
+	sectionRegexp := regexp.MustCompile(`### (New or )?Affected Resource\(s\)[^#]+`)
+	affectedResources := sectionRegexp.FindString(issueBody)
+	var labels []string
+	for label, resources := range enrolledTeams {
+		for _, resource := range resources {
+			if strings.Contains(affectedResources, resource) {
```
Comment (roaks3) [inline tools/issue-labeler/main.go:29]: It might be worth logging which resource was found at the time this was run, for debugging purposes

**Review by roaks3 — CHANGES_REQUESTED** [review]
Overall LGTM, but I didn't see any tests. Functionality like this should be very conducive for some quick unit tests, and it will help describe what is expected to happen (and also help prevent regressions).

File: tools/issue-labeler/main.go (hunk 1 above)
Comment (trodge) [reply tools/issue-labeler/main.go]: Done.

**Review by roaks3 — APPROVED** [review]
LGTM

**Pull request merged**